app/src/main/kotlin/com/exallium/todoapp/notedetail/NoteDetailPresenter.kt

```diff
@@ -0,0 +1,38 @@
+package com.exallium.todoapp.notedetail
+
+import com.exallium.todoapp.entities.Note
+import com.exallium.todoapp.mvp.BasePresenter
+import rx.Single
+import rx.SingleSubscriber
+import timber.log.Timber
+
+/**
+ * Presenter for Note Detail Screen
+ */
+class NoteDetailPresenter(val view: NoteDetailView, val model: NoteDetailModel) : BasePresenter<NoteDetailView>(view) {
+    lateinit var note: Note
+
+    override fun onViewCreated() {
+        if (view.isCached()) {
+            return
+        }
+        lookupNoteDetail(view.getNoteDetailId())
```

Comment:
How strongly do you feel about this? I don't fully understand your suggestion. Also, don't we want to avoid having model to deal with android stack (Bundle)?